addons/website_product_comparison/static/src/js/website_product_comparison.js

```diff
@@ -0,0 +1,342 @@
+odoo.define('website_product_comparison.comparison', function (require) {
+"use strict";
+
+var ajax = require('web.ajax');
+var core = require('web.core');
+var _t = core._t;
+var utils = require('web.utils');
+var Widget = require('web.Widget');
+var website = require('web_editor.base');
+
+if(!$('.oe_website_sale').length) {
+    return $.Deferred().reject("DOM doesn't contain '.oe_website_sale'");
+}
+
+var qweb = core.qweb;
+ajax.loadXML('/website_product_comparison/static/src/xml/comparison.xml', qweb);
+
+var ProductFeaturePanel = Widget.extend({
+    template: 'product_feature_template',
+    events: {},
+    product_data: {},
+    init: function() {},
+    show: function() {
+        $('.o_product_feature_panel').show();
+    },
+    load_products:function(product_ids) {
+        var self = this;
+        return ajax.jsonRpc('/shop/get_product_data', 'call', {
+            'product_ids': product_ids
+        }).then(function (data) {
+            _.each(data, function(product) {
+                self.product_data[product.product.id] = product;
+            });
+        });
+    },
+    animate_clone: function(cart, $elem, offset_top, offset_left) {
+        cart.find('.o_product_circle').addClass('o_red_highlight o_shadow_animation').delay(500).queue(function(){
+            $(this).removeClass("o_shadow_animation").dequeue();
+        });
+        var imgtodrag = $elem.find('img').eq(0);
+        if (imgtodrag.length) {
+            var imgclone = imgtodrag.clone()
+            .offset({
+                top: imgtodrag.offset().top,
+                left: imgtodrag.offset().left
+            })
+            .addClass('o_product_comparison_animate')
+            .appendTo($('body'))
+            .animate({
+                'top': cart.offset().top + offset_top,
+                'left': cart.offset().left + offset_left,
+                'width': 75,
+                'height': 75
+            }, 1000, 'easeInOutExpo');
+
+            imgclone.animate({
+                'width': 0,
+                'height': 0
+            }, function () {
+                $(this).detach();
+            });
+        }
+    }
+});
+
+var ProductComparison = Widget.extend({
+    template:"product_comparison_template",
+    events: {
+        'click .o_product_panel_header': 'toggle_panel',
+    },
+    init: function(parent){
+        this.parent = parent;
+        this.comparelist_product_ids = JSON.parse(utils.get_cookie('comparelist_product_ids') || 'null') || [];
```

Comment:
`JSON.parse(utils.get_cookie('comparelist_product_ids') || '[]')` but more generally, should be fault tolerant if invalid cookie data (e.g. delete the cookie)